runelite-client/src/main/java/net/runelite/client/plugins/microbot/mahoganyhomez/MahoganyHomesScript.java

```diff
@@ -0,0 +1,316 @@
+                    Rs2Dialogue.keyPressForDialogueOption(plugin.getConfig().currentTier().getPlankSelection().getChatOption());
+                    sleepUntil(() -> !Rs2Dialogue.hasSelectAnOption(), 5000);
+                    sleep(1200, 2200);
+                }
+
+            }
+
+        }
+
+    }
+
+    // Bank if we need to
+    private void bank() {
+        if(plugin.getCurrentHome() != null
+        && !plugin.getCurrentHome().isInside(Rs2Player.getWorldLocation())
+        && isMissingItems()) {
+            if(Rs2Bank.walkToBankAndUseBank()){
+                sleep(600,1200);
+                if(Rs2Inventory.getEmptySlots()-steelBarsNeeded() > 0)
+                    Rs2Bank.withdrawX(plugin.getConfig().currentTier().getPlankSelection().getPlankId(),Rs2Inventory.getEmptySlots()-steelBarsNeeded());
+                sleep(600,1200);
+                if(steelBarsNeeded()>steelBarsInInventory()) {
+                    Rs2Bank.withdrawX(ItemID.STEEL_BAR,steelBarsNeeded());
+                    sleep(600,1200);
+                }
+                Rs2Bank.closeBank();
+                return;
+            }
+            sleep(600,1200);
+        }
+    }
+
+    // Walk to current home
+    private void walkToHome() {
+        if(plugin.getCurrentHome() != null
+        && !plugin.getCurrentHome().isInside(Rs2Player.getWorldLocation())
+        && !isMissingItems()) {
+            Rs2Walker.walkWithState(plugin.getCurrentHome().getLocation(),3);
+        }
+    }
+
+    private boolean isMissingItems() {
+        return planksInInventory() < planksNeeded()
+                || steelBarsInInventory() < steelBarsNeeded();
+    }
+
+    private int planksNeeded() {
+        return plugin.getCurrentHome().getRequiredPlanks(plugin.getContractTier());
+    }
+
+    private int steelBarsNeeded() {
+        return plugin.getCurrentHome().getRequiredSteelBars(plugin.getContractTier());
+    }
+
+    private int planksInInventory() {
+        return Rs2Inventory.count(plugin.getConfig().currentTier().getPlankSelection().getPlankId());
+    }
+
+    private int steelBarsInInventory() {
+        return Rs2Inventory.count(ItemID.STEEL_BAR);
+    }
+
+    // Get closest contract location
+    private WorldPoint getClosestContractLocation() {
+        List<WorldPoint> contractLocations = new ArrayList<>();
+        contractLocations.add(ContractLocation.MAHOGANY_HOMES_ARDOUGNE.getLocation());
+        contractLocations.add(ContractLocation.MAHOGANY_HOMES_FALADOR.getLocation());
+        contractLocations.add(ContractLocation.MAHOGANY_HOMES_HOSIDIUS.getLocation());
+        contractLocations.add(ContractLocation.MAHOGANY_HOMES_VARROCK.getLocation());
+
+        return contractLocations.stream()
+                .min(Comparator.comparingInt(wp -> wp.distanceTo2D(Rs2Player.getWorldLocation())))
+                .orElse(null);
+    }
+
+    @Override
+    public void shutdown() {
+        mainScheduledFuture.cancel(true);
```

Comment:
This is done by script.java, it also checks if mainschedulefuture is not null before running the cancel method. So you can delete this line